crates/actors/src/block_tree_service.rs

```
@@ -2048,22 +2061,18 @@ impl BlockTreeCache {
 
     /// Prunes blocks below specified depth from tip. When pruning an on-chain block,
     /// removes all its non-on-chain children regardless of their height.
-    pub fn prune(&mut self, depth: u64) -> eyre::Result<()> {
-        if self.blocks.is_empty() {
-            return Ok(());
-        }
-
+    pub fn prune(&mut self, depth: u64) {
         let tip_height = self
             .blocks
             .get(&self.tip)
-            .ok_or_else(|| eyre::eyre!("Tip block not found"))?
+            .expect("Tip block not found")
```

Is it intended that this now panics with the expect() rather than bubbling the error up with an early return?